infrastructure/modules/platform/main.tf

```
@@ -97,3 +97,95 @@ resource "azurerm_application_insights" "app_insights" {
   resource_group_name = azurerm_resource_group.platform.name
   application_type    = "other"
 }
+
+resource "random_password" "password" {
+  length           = 32
+  special          = true
+  override_special = "_%@"
+}
+
+resource "random_string" "server" {
+  length  = 15
+  special = false
+}
+
+data "azurerm_client_config" "current" {
+}
+
+resource "azurerm_key_vault" "secret_vault" {
+  name                = format("%s%s", "kv-fnhs", random_string.server.result)
```

this is equivalent to:
```suggestion
  name                = "kv-fnhs${random_string.server.result}"
```

although I would suggest that `server` is not a great name for this, and it might benefit from including the environment name. Maybe:

```suggestion
  name                = "fnhs-keyvault-${var.environment}-${random_string.keyvault_suffix.result}"
```

---

I wanted this also, but there is a character limit of 24 alphanumeric chars, and the name needs to be completely unique inside the whole of Azure.  This includes recently deleted names.
So if you created: 
fnhs-keyvault-dev-david${random_string.keyvault_suffix.result}
The random string would only be 1 character, which would give you a lower opportunity to get a unique name.  Also, "david" is 5 chars, so we don't have a way in terraform to chop part of a name off if someone has a name like "declan".

---

(also avoiding `kv` because `kv` means `key-value` to me, so I would expect this to be a redis instance or something).